### DISCUSSION TITLE
G7 Loses Pairing once every 24 hours or so

### DISCUSSION THREAD

**Discussion by mikeloeven** [NightscoutFoundation/xDrip#3274] [Bug reports]
XDrip Version  641cc94 Jan 2 2024 Build 
Pixel 8: Graphene OS 14: Build UQ1A.231205.015.2023123100

Recently switched from Dexcom G6 to G7 unit. I was able to get the G7 connected however the sensor seems to become unpaired at least once every 24 hour period or so and needs to be manually reconnected. any workaround for this?

**Comment by Navid200** [comment]
Pixel 8?  That phone shouldn't have any such problems.
Have you set battery optimization to unrestricted?
Do you mind posting a screenshot of the following page?
Settings &#8722;> G5/G6/Dex1 Debug Settings

**Reply by mikeloeven** [reply]
When I went to check the debug settings I found a setting marked as a workaround for devices that periodically lose their pairing. I tried enabling that option and looks like the the issue has resolved at least I have not had a pairing loss for the last few days now. If you still want any logs or screenshots for development purposes i can provide them.

**Reply by Navid200** [reply]
No, if it is working fine, that's OK.
But, I didn't expect Pixel 8 to need that setting enabled.

**Reply by mikeloeven** [reply]
I am running Graphene OS not stock android so could be something with the enhanced network security stuff causing the issue Its pretty paranoid about anything that requests location even if its just for bluetooth scan
